kolibri/plugins/learn/assets/test/bookmark-page.spec.js

```diff
@@ -0,0 +1,51 @@
+import { shallowMount } from '@vue/test-utils';
+import BookmarkPage from '../src/views/BookmarkPage';
+
+describe('Bookmark page', () => {
+  let wrapper;
+  let loadMoreSpy;
+  let createdSpy;
+  let removeFromBookmarksSpy;
+
+  beforeAll(() => {
+    loadMoreSpy = jest.spyOn(BookmarkPage.methods, 'loadMore');
+    createdSpy = jest.spyOn(BookmarkPage, 'created').mockImplementation(() => Promise.resolve());
+    removeFromBookmarksSpy = jest.spyOn(BookmarkPage.methods, 'removeFromBookmarks');
+
+    wrapper = shallowMount(BookmarkPage, {
+      stubs: {
+        ContentCardGroupGrid: {
+          name: 'ContentCardGroupGrid',
+          template: '<div></div>',
+        },
+      },
+    });
+    wrapper.setData({
+      loading: false,
+      more: true,
+      bookmarks: [{ a: 'b' }],
+    });
+  });
+
+  it('smoke test', () => {
+    const wrapper = shallowMount(BookmarkPage);
+    expect(createdSpy).toHaveBeenCalled();
+    expect(wrapper.exists()).toBe(true);
+  });
+  describe('When the user clicks the remove from bookmarks icon', () => {
+    it('will make a call to remove the bookmark from the list of bookmarks', () => {
+      wrapper.findComponent({ name: 'ContentCardGroupGrid' }).vm.$emit('removeFromBookmarks');
+      expect(removeFromBookmarksSpy).toHaveBeenCalled();
```

Comment:
This doesn't test if a call was made. If there was an error in `removeFromBookmarks` method, this test would still pass even though the call wasn't made. That's one of the reasons why it's usually not recommended to test internal methods of components.

I'd recommend mocking `client` and checking that it's been called with correct `method` and `url` parameters.